dev/com.ibm.ws.cdi.1.2_fat/fat/src/com/ibm/ws/cdi12/suite/FATSuite.java

```diff
@@ -0,0 +1,1341 @@
+                AppClientTest.class,
+                AppClientAdvancedTest.class,
+                AppClientSecurityTest.class,
+                AppExtensionTest.class,
+                BeanDiscoveryModeNoneTest.class,
+                BeanLifecycleTest.class,
+                BeanManagerLookupTest.class,
+                CDI12JNDIBeanManagerTest.class,
+                CDI12ExtensionTest.class,
+                CDI12WebServicesTest.class,
+                CDICurrentTest.class,
+                ClassExclusionTest.class,
+                ClassLoadPrereqLogger.class,
+                ClassMaskingTest.class,
+                ConversationFilterTest.class,
+                DecoratorOnBuiltInBeansTest.class,
+                DeltaSpikeSchedulerTest.class,
+                DisablingBeansXmlValidationTest.class,
+                DynamicBeanExtensionTest.class,
+                EjbConstructorInjectionTest.class,
+                EjbDiscoveryTest.class,
+                EjbMiscTest.class,
+                EJB32Test.class,
+                EjbTimerTest.class,
+                EmptyCDITest.class,
+                EnablingBeansXmlValidationTest.class,
+                EventMetaDataTest.class,
+                GloballyEnableUsingPriorityTest.class,
+                ImplicitBeanArchivesDisabledTest.class,
+                ImplicitBeanArchiveNoAnnotationsTest.class,
+                ImplicitBeanArchiveTest.class,
+                ImplicitWarLibJarsTest.class,
+                ImplicitWarTest.class,
+                InjectInjectionPointTest.class,
+                InjectParameterTest.class,
+                JarInRarTest.class,
+                JEEInjectionTargetTest.class,
+                JNDILookupTest.class,
+                CDIManagedBeanInterceptorTest.class,
+                MultipleBeansXmlTest.class,
+                MultiModuleAppTest.class,
+                MultipleNamedEJBTest.class,
+                NonContextualInjectionPointTest.class,
+                NonContextualTests.class,
+                ObservesInitializedTest.class,
+                PackagePrivateAccessTest.class,
+                PassivationBeanTests.class,
+                RootClassLoaderTest.class,
+                SessionDestroyTests.class,
+                SharedLibraryTest.class,
+                SessionDestroyTests.class,
+                SimpleJSFTest.class,
+                SimpleJSFWithSharedLibTest.class,
+                SimpleJSPTest.class,
+                StatefulSessionBeanInjectionTest.class,
+                ValidatorInJarTest.class,
+                VisTest.class,
+                WarLibsAccessWarBeansTest.class,
+                WebBeansBeansXmlInWeldTest.class,
+                WithAnnotationsTest.class
+})
+public class FATSuite {
+    
+    private static final Logger LOG = Logger.getLogger(FATSuite.class.getName());
+    private static Map<String,List<Archive>> serversToApps = new HashMap<String,List<Archive>>();
+
+    static{
+        
+
```

Comment:
I thought that you were going to come up with a clever way to so this ShrinkWrap stuff automatically rather than hard coding it? If you are hard-coding, is there some way to split it up and organise better for when we need to make changes or add new apps?

Reply:
This stuff is all static, can it live in static methods in each test class?

Reply:
The gradle scripts don't preserve any information about which class goes in which app once it's transfered to the autoFVT directory, so I told my code to output shrinkwrap code and copied and pasted it into here. 

This way is less clever, and less pretty, but it is better since it means the tests don't need to walk the directory tree every time we run the tests. For new tests we can write them in a shrinkwrap safe way (package names guarenteed to be unique to the app) and use the default app stuff in the shrink wrap helper class which will only need a couple of lines of Shrinkwrap.